apps/api_web/lib/api_web/controllers/portal/user_controller.ex

```diff
@@ -20,18 +20,47 @@ defmodule ApiWeb.ClientPortal.UserController do
     |> render("new.html", changeset: changeset)
   end
 
+  def create(conn, %{"user" => user_params, "g-recaptcha-response" => recaptcha}) do
+    _create(conn, user_params, recaptcha)
+  end
+
+  # Allow missing recaptcha response
   def create(conn, %{"user" => user_params}) do
-    case ApiAccounts.register_user(user_params) do
-      {:ok, user} ->
-        conn
-        |> put_session(:user_id, user.id)
-        |> configure_session(renew: true)
-        |> redirect(to: portal_path(conn, :index))
+    _create(conn, user_params, nil)
+  end
 
+  defp _create(conn, user_params, recaptcha) do
+    # Only test recaptcha if the feature is enabled (prod):
+    recaptcha_verified =
+      if Application.get_env(:recaptcha, :enabled) do
+        Recaptcha.verify(recaptcha)
+      else
+        {:ok, False}
```

Comment:
suggestion: you probably meant `false` here.
```suggestion
        {:ok, false}
```
However, I'm a little confused about how `false` would be the correct return value here. What does `Recaptch.verify/1` normally return?